src/coloring.c

```diff
@@ -0,0 +1,107 @@
+
+#include "igraph_coloring.h"
+#include "igraph_interface.h"
+#include "igraph_adjlist.h"
+#include "igraph_interrupt_internal.h"
+#include <assert.h>
+
+
+int igraph_i_vertex_coloring_greedy_cn(const igraph_t *graph, igraph_vector_int_t *colors) {
```

Comment:
Return type should be `igraph_error_t`.

Reply:
Most existing functions use `int` as the return type. `igraph_error_type_t` is an enum, and my compiler represents it with an unsigned value. This then causes a large number of warnings due to signedness conversion in `IGRAPH_CHECK`.

Question: Do any of the functions that return an error code ever return a negative value? (Of course, that wouldn't correspond to any existing enum value, but that doesn't mean that it's never used...)  If not, it would be nice to change `IGRAPH_CHECK` to use `igraph_error_type_t` internally, and eliminate some of the warnings.

Reply:
Okay, just stick to `int` for the time being - I have forgotten that we weren't using the enum for the return value, sorry.